packages/ui/src/views/layouts/main-content-layout.tsx

```diff
@@ -1,49 +1,58 @@
-import { CSSProperties, PropsWithChildren, ReactNode } from 'react'
+import { PropsWithChildren, ReactNode } from 'react'
 
+import { SidebarContext } from '@/components'
 import { useTheme } from '@/context'
-import { cn, wrapConditionalObjectElement } from '@/utils'
-import { ScrollArea, ScrollAreaProps } from '@components/scroll-area'
+import { cn } from '@/utils'
 
-interface MainContentLayoutProps extends PropsWithChildren<unknown> {
-  className?: string
+const HalfArch = ({ className }: { className?: string }) => (
+  <div className="relative size-1.5 overflow-hidden">
+    <div
+      className={cn(
+        'absolute top-0 size-3 rounded-full border shadow-[0_0_0_20px_hsl(var(--canary-sidebar-background-01))]',
```

Comment:
should we be using a standard shadow here, or create a tailwind token instead of custom inline shadow?

Reply:
No need for a token here because it's not a usual shadow - it's just a part of the trick that is used for the `Inset` theme only: it creates an arch overlay that keeps the top corners of the content rounded when the page is scrolled. I'll add a comment above the component